src/chart.zoom.js

```diff
@@ -201,8 +201,13 @@ function panIndexScale(scale, delta, panOptions) {
 
 function panTimeScale(scale, delta, panOptions) {
 	var options = scale.options;
-	options.time.min = rangeMinLimiter(panOptions, scale.getValueForPixel(scale.getPixelForValue(scale.firstTick) - delta));
-	options.time.max = rangeMaxLimiter(panOptions, scale.getValueForPixel(scale.getPixelForValue(scale.lastTick) - delta));
+	
+	var newMax = rangeMaxLimiter(panOptions, scale.getValueForPixel(scale.getPixelForValue(scale.lastTick) - delta));
+	var newMin = rangeMinLimiter(panOptions, scale.getValueForPixel(scale.getPixelForValue(scale.firstTick) - delta));
+	if (newMax != options.time.max && newMin != options.time.min) {
```

Comment:
I was just thinking something like:

```javascript
newMax <= options.time.max && newMin >= options.time.min
```

To address the undefined issue:

```javascript
(!options.time.max || newMax <= options.time.max) && (!options.time.min || newMin >= options.time.min)
```

I haven't tested this, FYI.